R/datasets.R

```diff
@@ -650,3 +660,39 @@ promote_headers_behavior <- function(option) {
   option <- as.integer(option)
   azureml$data$dataset_type_definitions$PromoteHeadersBehavior(option)
 }
+
+#' Represents a path to data in a datastore.
+#'
+#' @description
+#' The path represented by DataPath object can point to a directory or a data artifact (blob, file). 
+#'
+#' @param datastore The Datastore to reference.
+#' @param path_on_datastore The relative path in the backing storage for the data reference.
+#' @param name An optional name for the DataPath.
+#' @return The `DataPath` object.
+#' @export
+#' @examples
+#' \dontrun{
+#' my_data <- register_azure_blob_container_datastore(
+#'     workspace = ws, 
+#'     datastore_name = blob_datastore_name, 
+#'     container_name = ws_blob_datastore$container_name, 
+#'     account_name = ws_blob_datastore$account_name, 
+#'     account_key = ws_blob_datastore$account_key, 
+#'     create_if_not_exists = TRUE)
+#'
+#' datapath <- data_path(my_data, <path_on_my_datastore>)
+#' dataset <- create_file_dataset_from_files(datapath)
+#' }
+#' @seealso
+#' \code{\link{create_file_dataset_from_files}}
+#' \code{\link{create_tabular_dataset_from_parquet_files}}
+#' \code{\link{create_tabular_dataset_from_delimited_files}}
+#' \code{\link{create_tabular_dataset_from_json_lines_files}}
+#' \code{\link{create_tabular_dataset_from_sql_query}}
+#' @md
+data_path <- function(datastore = NULL, path_on_datastore = NULL, name = NULL) {
```

Comment:
should we remove `= NULL` from datastore argument so it is clear that this parameter is required?

Reply:
I was going to ask you about that. I'll remove it.